src/lib/support/Span.h

```diff
@@ -36,8 +37,16 @@ class Span
     constexpr explicit Span(const T (&databuf)[N]) : Span(databuf, N)
     {}
 
+    bool operator==(const Span<T> & other) const
```

Comment:
I'm not sure we should name this operation `operator==`.  In some sense, `operator==` means the two objects are "the same", such that if `A == B` then doing `A = B` should be more or less a no-op.  That is decidedly _not_ the case with `Span`.

Perhaps this should be called `HasSameData` or some such?

Reply:
how about data_equal() ?

Reply:
@bzbarsky-apple please take a look